.github/workflows/engine-checks.yml

```diff
@@ -234,9 +238,9 @@ jobs:
         run: ./run --help || (git clean -ffdx && ./run --help)
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - if: "${{ (contains(github.event.pull_request.labels.*.name, 'CI: Clean build required') || (github.ref == 'refs/heads/develop') || inputs.clean_build_required) }}"
+      - if: ${{ (always()) }}
```

Comment:
I am still convinced that this job should not always use clean before. See my other comment in https://github.com/enso-org/enso/pull/14213/files#r2524081982.